ta4j-core/src/main/java/org/ta4j/sparx/criteria/CompoundedAnnualGrowthReturnCriteria.java

```diff
@@ -0,0 +1,41 @@
+package org.ta4j.sparx.criteria;
+
+import org.ta4j.core.BarSeries;
+import org.ta4j.core.Position;
+import org.ta4j.core.TradingRecord;
+import org.ta4j.core.criteria.AbstractAnalysisCriterion;
+import org.ta4j.core.criteria.pnl.ReturnCriterion;
+import org.ta4j.core.num.DoubleNum;
+import org.ta4j.core.num.Num;
+
+public class CompoundedAnnualGrowthReturnCriteria extends AbstractAnalysisCriterion {
+    @Override
+    public Num calculate(BarSeries series, Position position) {
+        Num grossReturn = new ReturnCriterion().calculate(series, position);
+        double cagr = calculateCagr(series, grossReturn);
+        return DoubleNum.valueOf(cagr);
```

Comment:
Use `series.NumOf()` rather than `DoubleNum.valueOf()` as it maintains a standard Num implementation throughout execution.